File: runningSum.py
def runningSum(nums): # TC O(n) // SC O(1)
	for i in range(1,len(nums)):
		nums[i]+=nums[i-1]
	return nums 

def runningSum1(nums): # TC O(n) // SC O(n)
	dp=[0]*len(nums)
	dp[0]=nums[0]
	for i in range(1,len(nums)):
		dp[i]=dp[i-1]+nums[i]
	return dp 

def runningSum2(nums):
	return [sum(nums[:i+1]) for i in range(len(nums))]

print(runningSum([1,2,3,4]))
print(runningSum1([1,1,1,1,1]))
print(runningSum2([3,1,2,10,1]))

File: runningSum.java
import java.util.*;
// Approach 1- Bottom-Up DP 
// The very base DP problem
class Solution { // TC O(n) SC O(n)
    public int[] runningSum(int[] nums) {
        int[] ans=new int[nums.length];
        ans[0]=nums[0];
        for(int i=1; i<nums.length; i++){
            ans[i]=nums[i]+ans[i-1];
        }
        return ans;
    }
}

class runningSum{
    public static void main(String[] args) {
        int[] nums={1,2,3,4,5};
        Solution s= new Solution();
        System.out.println(Arrays.toString(s.runningSum(nums)));
    }
}